packages/components-native/src/Form/Form.tsx

```diff
@@ -245,6 +245,19 @@ function InternalForm<T extends FieldValues, S>({
     setKeyboardScreenY(0);
   }
 
+  function callHandleSubmit() {
+    let result: S;
+
+    handleSubmit(data => {
+      internalSubmit(data).then(saveResult => {
+        result = saveResult as S;
+      })
+    })().then(() => {
+          removeListenerRef.current?.();
+          onSubmitSuccess(result);
```

Comment:
Then method only being called after the form has acknowledged a successful save. May need to add checks for errors in form state.